### ISSUE TITLE
Uppercase characters are not allowed in package names

### ISSUE THREAD

**Issue by joeroe** [benmarwick/rrtools#76]
`create_compendium()` and `use_compendium()` refuse to make a package with uppercase characters in the name:

```
> rrtools::use_compendium("LongWindedScientificAnalysis")
Error: 'LongWindedScientificAnalysis' is not a valid package name. It should:
* Contain only ASCII letters, numbers, and '.'
* Have at least two characters
* Start with a letter
* Not end with '.'
* Not contain any upper case characters
```

This was added f2e3b43, to ensure compatibility with travis. But I'd make the case for allowing uppercase characters:

* R, CRAN, and GitHub all support mixed case. Only travis is a problem, but most<sup>?</sup> users won't use travis.
* Research compendiums are more likely to have long, multi-word names than conventional packages. These are hard to parse in all-lowercase (longwindedscientificanalysis).
* Common word separators like `_` and `-`—which would solve the above problem—aren't allowed in R package names. `.` is, but most modern R style guides recommend against using it as a word separator.

A solution would be to throw a warning about the travis issue when creating a compendium with uppercase characters, rather than an error. The hard check could be moved to `use_travis()`.

**Comment by benmarwick** [comment]
Yes, a warning sounds like a good alternative to a strict rule, thanks for the suggestion. 

Another option here is to separate the package name from the repo name. For example @cboettig names his compendium pkgs simply 'compendium', and then has more flexible names for the repo: https://github.com/cboettig/noise-phenomena/blob/master/DESCRIPTION

I like that because it saves the user from making a decision they probably don't care about. CRAN is not the end point for these compendium pkgs, so it hardly matters what the pkg is called. 

What do you think about that?

**Comment by joeroe** [comment]
That sounds like a good idea. I can think of some reasons why you might want the package to have a descriptive name, e.g. if you want to library() it in another project, so maybe add it as an option rather than a default?

**Comment by ha0ye** [comment]
Chiming in to say:
* I would also like to use uppercase characters in package names.
* I think package names that differ from repo names can lead to naming confusion. I think it would be good to encourage them to be the same name, but users can always bypass by renaming the repo manually, right?

**Comment by benmarwick** [comment]
Thanks for the feedback, we couldn't remember why we had this rule, and everything seems to work fine with upper and lower case mixed together, as you say. So we've removed that rule in https://github.com/benmarwick/rrtools/commit/caeb97f03772c5d3ce0ac71240c2315de147eeab

![image](https://user-images.githubusercontent.com/1262179/56184138-f67f8f00-5fcd-11e9-8aa3-03af66a1dd16.png)

Thanks again!
